src/jit/compiler.cpp

```diff
@@ -11085,3 +11085,231 @@ bool Compiler::killGCRefs(GenTree* tree)
     }
     return false;
 }
+
+//------------------------------------------------------------------------
+// getSiVarLoc: Returns a "siVarLoc" instance representing the place where the variable 
+// is given its description, "baseReg", and "offset" (if needed).
+//
+// Arguments:
+//    varDsc    - a "LclVarDsc *" to the variable it is desired to build the "siVarLoc".
+//    baseReg   - a "regNumber" use as a base for the offset.
+//    offset    - a signed amount of bytes distance from "baseReg" for the position of the variable.
+//
+// Return Value:
+//    A "siVarLoc" filled with the correct case struct fields for the variable, which could live
+//    in a register, an stack position, or a combination of both.
+//
+// Assumptions:
+//    No new memory is allocated on this method.
+//    No changes on "varDsc" are applied.
+//
+// Notes:
+//    The "baseReg" and "offset" cannot be obtained from "varDsc".
+//    It calls "fillRegisterVarLoc" or "fillStackVarLoc" depending on if the "varDsc" argument.
+//    "lvIsInReg", so their notes and assumptions are held.
+Compiler::siVarLoc Compiler::getSiVarLoc(LclVarDsc* varDsc, regNumber baseReg, signed offset) const
+{
+    siVarLoc  varLoc;
+    var_types type = genActualType(varDsc->TypeGet());
+
+    if (varDsc->lvIsInReg())
+    {
+        fillRegisterVarLoc(varDsc, varLoc, type, baseReg, offset);
+    }
+    else
+    {
+        fillStackVarLoc(varDsc, varLoc, type, baseReg, offset);
+    }
+
+    return varLoc;
+}
+
+//------------------------------------------------------------------------
+// fillStackVarLoc: Fill "varLoc" argument indicating the stack position of the variable
+// using "LclVarDsc" and "baseReg"/"offset".
+//
+// Arguments:
+//    varDsc    - a "LclVarDsc *" to the variable it is desired to build the "siVarLoc".
+//    varLoc    - a "siVarLoc &" to fill with the data of the "varDsc".
+//    type      - a "var_types" which indicate the type of the variable.
+//    baseReg   - a "regNumber" use as a base for the offset.
+//    offset    - a signed amount of bytes distance from "baseReg" for the position of the variable.
+//
+// Assumptions:
+//    No new memory is allocated on this method.
+//    No changes on "varDsc" are applied.
+//
+// Notes:
+//    The "varLoc" argument is filled depending of the "type" argument but as a VLT_STK... variation.
+//    A Exception is thrown if type doesn't match with any of the listed on the switch.
```

Comment:
This code doesn't throw any exceptions. Well, at least nobody refers to `noway_assert` as "exception".